### PR TITLE
Implemented additional instructions and examples

### PR DESCRIPTION
**Pull request by jordanhudgens**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,18 +93,38 @@ The `form_for` above will render the following HTML:
...
+* `form_for` handles the retrieval of values from your object model, and will also try to route the form to the appropriate action specified in the controller.
+
+So when would you choose one over the other? Below are some real world examples:
+
+* `form_for` - this works well for forms that manage CRUD. Imagine that you have a blog posting application, a great fit for the `form_for` method would be the `Post` model. The reason is because the `Post` model would have the standard `ActiveRecord` setup, and therefore it's smart to take advantage of the functionality built into `form_for`.
+
+* `form_tag` - this works well for forms that are not directly connected with models. For example, let's say that our blog posting application has a search engine, the search form would be a great fit for using `form_tag`.
+
 
 ## Fields_for
```
Comment (jmburges) [inline README.md]: Actually can you yank out this stuff on fields_for? They haven't covered any of nested_attributes so let's just cover it later

File: README.md [hunk 2]
```diff
@@ -81,19 +135,22 @@ The `form_for` above will render the following HTML:
...
-### collection_check_boxes
-- It assumes that there is a many-to-many relationship in place (books have many authors, authors have many books)
-- Typically used with a form builder element in a `form_for`
+### `collection_check_boxes`
+
+- `collection_check_boxes` assumes that there is a `many-to-many` relationship in place. Let's imagine that you have a book publishing application: `books` have many `authors`, `authors` have many `books`
+
+- `collection_check_boxes` typically are used with a form builder element in a `form_for`
+
+Below is an example on how you can implement `collection_check_boxes`:
```
Comment (jmburges) [inline README.md:124]: Really quickly, can you show what HTML this renders?

File: README.md [hunk 3]
```diff
@@ -112,9 +169,11 @@ The `form_for` above will render the following HTML:
   - The third parameter is `:id`: this is the parameter of an author that will be passed into the `:author_ids` collection once a checkbox is checked
   - The final parameter is `:name`: this is the attribute of author that will be rendered on the page next to the checkbox
 
-### collection_select
-- In this case, this form assumes that there is a one-to-many relationship in place (movie has one director, director has many movies)
-- Used in this case with a form builder element in a `form_for` helper
+### `collection_select`
+
+- `collection_select` assumes that there is a one-to-many relationship in place. Imagine that you are creating a competitor for IMDB.com, you would have the model of a `movie` that has one `director`, and the `director` model would be setup to have many movies
```
Comment (jmburges) [inline README.md:159]: Can you show what HTML this renders?

**Comment by jordanhudgens** [comment]
@jmburges This has been updated.

**Pull request merged**